src/main/java/com/petros/bibernate/dao/EntityPersister.java

```diff
@@ -58,12 +68,65 @@ public <T> List<T> findAll(Class<T> entityClass, Field field, Object fieldValue)
         return result;
     }
 
+    public <T> T insert(T entity) {
+        Objects.requireNonNull(entity);
+        try (var connection = dataSource.getConnection()) {
+            PreparedStatement insertStatement = prepareInsertStatement(entity, connection);
+
+            int rowsAffected = insertStatement.executeUpdate();
+            if (rowsAffected != 1) {
+                throw new BibernateException("Failed to insert entity into the database");
```

Comment:
Maybe extraction to separate method (requireOnlyOneAffectedRow) would be a little more readable.